packages/legend-application-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -693,6 +707,268 @@ export class LocalChangesState {
           true,
         ),
       ]);
+
+      this.editorStore.applicationStore.log.info(
+        LogEvent.create(CHANGE_DETECTION_EVENT.CHANGE_DETECTION_RESTARTED),
+        Date.now() - syncFinishedTime,
+        'ms',
+      );
+      // ======= FINISHED (RE)START CHANGE DETECTION =======
+    } catch (error) {
+      assertErrorThrown(error);
+      this.editorStore.applicationStore.log.error(
+        LogEvent.create(LEGEND_STUDIO_APP_EVENT.SDLC_MANAGER_FAILURE),
+        error,
+      );
+      if (
+        error instanceof NetworkClientError &&
+        error.response.status === HttpStatus.CONFLICT
+      ) {
+        // NOTE: a confict here indicates that the reference revision ID sent along with update call
+        // does not match the HEAD of the workspace, therefore, we need to prompt user to refresh the application
+        this.editorStore.applicationStore.notifyWarning(
+          'Syncing failed. Current workspace revision is not the latest. Please backup your work and refresh the application',
+        );
+        // TODO: maybe we should do more here, e.g. prompt the user to download the patch, but that is for later
+      } else {
+        this.editorStore.applicationStore.notifyError(error);
+      }
+    } finally {
+      this.pushChangesState.complete();
+    }
+  }
+}
+
+export class TextLocalChangesState extends LocalChangesState {
+  localChanges: EntityChange[] = [];
+
+  constructor(editorStore: EditorStore, sdlcState: EditorSDLCState) {
+    super(editorStore, sdlcState);
+    makeObservable(this, {
+      setLocalChanges: action,
+    });
+  }
+
+  setLocalChanges(val: EntityChange[]): void {
+    this.localChanges = val;
+  }
+
+  *refreshLocalChanges(): GeneratorFn<void> {
+    const startTime = Date.now();
+    this.refreshLocalChangesDetectorState.inProgress();
+    try {
+      // ======= (RE)START CHANGE DETECTION =======
```

Comment:
dont think we use the change detection state at all here so should we still have that comment here ?